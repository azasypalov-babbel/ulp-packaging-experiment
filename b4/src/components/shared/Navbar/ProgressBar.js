import React from 'react';
import PropTypes from 'prop-types';
import cx from '../../../lib/cxHelper';

const getTrainerProgressStyles = ({
  hasPassed,
  isHead,
  sequenceHeadProgress
}) => {
  if (hasPassed) return 100;
  if (isHead) return sequenceHeadProgress * 100;
  return 0;
};

const ProgressBar = ({
  trainerCount,
  sequenceHeadIndex,
  sequenceHeadProgress,
  onTrainerClick
}) => (
  <div className={cx('progress-bar')}>
    <div className={cx('progress-bar__base')}>
      {Array(trainerCount).fill().map((trainer, index) => {
        const hasPassed = index < sequenceHeadIndex;
        const isHead = index === sequenceHeadIndex;

        return (
          <div
            key={`trainer-${index}`}
            onClick={() => onTrainerClick(index)}
            style={{ flexGrow: 1 }}
          >
            <span
              className={cx('progress-bar__item')}
              style={{ width: `${getTrainerProgressStyles({ hasPassed, isHead, sequenceHeadProgress })}%` }}
            >
            </span>
          </div>
        );
      })}
    </div>
  </div>
);

export const progressBarPropTypes = {
  trainerCount: PropTypes.number,
  sequenceHeadIndex: PropTypes.number,
  sequenceHeadProgress: PropTypes.number,
  onTrainerClick: PropTypes.func.isRequired
};

ProgressBar.propTypes = progressBarPropTypes;

export default ProgressBar;
