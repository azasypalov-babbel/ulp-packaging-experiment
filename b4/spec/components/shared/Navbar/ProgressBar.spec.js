import React from 'react';
import ProgressBar from '../../../../src/components/shared/Navbar/ProgressBar';
import { shallow } from 'enzyme';

describe('<ProgressBar />', () => {
  let component;

  const onTrainerClickMock = jest.fn();

  const props = {
    trainerCount: 4,
    sequenceHeadIndex: 2,
    currentTrainerIndex: 2,
    sequenceHeadProgress: 0.33,
    onTrainerClick: onTrainerClickMock
  };

  beforeEach(() => {
    component = shallow(<ProgressBar {...props} />);
  });

  test('it renders', () => {
    expect(component).toMatchSnapshot();
  });

  test('it calls `onTrainerClick` when clicking on progressbar', () => {
    const component = shallow(<ProgressBar {...props} />);
    const progressBarSegment = component.findWhere((node) => node.key() === 'trainer-0').first();

    progressBarSegment.simulate('click');
    expect(onTrainerClickMock).toHaveBeenCalled();
  });
});
