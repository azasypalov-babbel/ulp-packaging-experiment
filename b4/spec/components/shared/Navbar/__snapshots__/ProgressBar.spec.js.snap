// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> it renders 1`] = `
<div
  className="loy-progress-bar"
>
  <div
    className="loy-progress-bar__base"
  >
    <div
      key="trainer-0"
      onClick={[Function]}
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <span
        className="loy-progress-bar__item"
        style={
          Object {
            "width": "100%",
          }
        }
      />
    </div>
    <div
      key="trainer-1"
      onClick={[Function]}
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <span
        className="loy-progress-bar__item"
        style={
          Object {
            "width": "100%",
          }
        }
      />
    </div>
    <div
      key="trainer-2"
      onClick={[Function]}
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <span
        className="loy-progress-bar__item"
        style={
          Object {
            "width": "33%",
          }
        }
      />
    </div>
    <div
      key="trainer-3"
      onClick={[Function]}
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <span
        className="loy-progress-bar__item"
        style={
          Object {
            "width": "0%",
          }
        }
      />
    </div>
  </div>
</div>
`;
